nltk/translate/util.py

```diff
@@ -0,0 +1,144 @@
+''' UTILITY Functions for ROUGE score for Machine Translation'''
+
+from __future__ import division
+
+def jacknifing(score_list, averaging=True):
+        ''' This is a averaging function for calculating ROUGE score
+    when multiple references are present. In this technique, from a 
+    given list of m values, m unique sets of m-1 elements is created.
+    Then the maximum is calculated from each of this m sets. The 
+    averaging is then performed on the m maximum values thus obtained.
+    This technique has been referred to as the Jacknifing technique 
+    in the original paper that introduces ROUGE-SCORE.
+
+
+    param score_list : list of scores over which averaging occurs
+    type (score_list) : list
+
+    param averaging : Jacknifing occurs if averaging is True
+    type (averaging) : boolean
+
+    >>> scores=[10, 20, 30, 40]
+    
+    >>> jacknifing(scores, averaging=False)
+    [10, 20, 30, 40]
+
+    >>> jacknifing(scores, averaging=True)
+    37.5
+
+    >>> scores =[10]
+    
+    >>> jacknifing(scores)
+    10.0
+
+
+    '''
+        if(len(score_list) == 1):
+            return sum(score_list)/len(score_list)
+        elif((len(score_list) > 1) and (averaging is False)):
+            return score_list
+        else:
+            '''
+            average : store the maximum scores
+            from the m different sets of m-1 scores.
+            such that m is the len(score_list)
+            '''
+            average = []
+            for i in score_list:
+                # dummy : list a particular combo of m-1 scores
+                dummy = [j for j in score_list if i != j]
+                average.append(max(dummy))
+            return sum(average)/len(average)
+
+
+def f(k,  alpha=2, inverse=True):
+    '''Weighting function of the Longest-Common-Subsequence
+
+    param k : parameter entered by user
+    type (k) : float
+
+    param alpha : parameter entered by user
+    type(alpha) : float
+    
+    param inverse : parameter entered by user which decides whether 
+                    to return the function or it's inverse
+    type(inverse) : boolean                
+
+    '''
+    
+    return k**(1/alpha) if inverse else k**alpha
```

Comment:
Maybe there's no need for the inverse if the alpha is parametric. Simply allow the alpha to take in negative values. 

But I understand that the original rouge paper is particular with having alpha being `[0,  ∞)` and then describing the inverse from Section 4 of http://www.aclweb.org/anthology/W04-1013 

Nothing to change here but good to note.